app/src/main/java/net/vonforst/evmap/api/openstreetmap/OpenStreetMapModel.kt

```diff
@@ -0,0 +1,191 @@
+     * Return the name for this charging station.
+     */
+    private fun getName(): String {
+        // Ideally this station has a name.
+        // If not, fall back to the operator.
+        // If that is missing as well, use a generic "Charging Station" string.
+        return tags["name"]
+            ?: tags["operator"]
+            ?: "Charging Station";
+    }
+
+    /**
+     * Return the chargepoints for this charging station.
+     */
+    private fun getChargepoints(): List<Chargepoint> {
+        // Note: In OSM, the chargepoints are mapped as "socket:<type> = <count>"
+        val chargepoints = mutableListOf<Chargepoint>()
+        for (socket in SOCKET_TYPES) {
+            val count = try {
+                (this.tags[socket.osmSocketBaseTag()] ?: "0").toInt()
+            } catch (e: NumberFormatException) {
+                0
+            }
+            if (count > 0) {
+                if (socket.evmapKey != null) {
+                    val outputPower = parseOutputPower(this.tags["${socket.osmSocketBaseTag()}:output"])
+                    chargepoints.add(Chargepoint(socket.evmapKey, count, outputPower))
+                }
+            }
+        }
+        return chargepoints
+    }
+
+    private fun getOpeningHours(): OpeningHours? {
+        val rawOpeningHours = tags["opening_hours"] ?: return null
+
+        // Handle the simple 24/7 case
+        if (rawOpeningHours == "24/7") {
+            return OpeningHours(true, null, null)
+        }
+
+        // TODO: Try to convert other formats as well?
+        //
+        // Note: The current {@link OpeningHours} format is not flexible enough to handle
+        // all rules that OSM can represent and might need to be updated.
+        // This library could help:
+        //
+        // Alternatively, with the opening-hours-evaluator library
+        // https://github.com/leonardehrenfried/opening-hours-evaluator
+        // we could implement an "open now" feature.
+        return null
+    }
+
+    private fun getCost(): Cost? {
+        val freecharging = when (tags["fee"]?.lowercase()) {
+            "yes", "y" -> false
+            "no", "n" -> true
+            else -> null
+        }
+        val freeparking = when (tags["parking:fee"]?.lowercase()) {
+            "no", "n" -> true
+            "yes", "y", "interval" -> false
+            else -> null
+        }
+        return Cost(freecharging, freeparking)
+    }
+
+    companion object {
+        /**
+         * Parse raw OSM output power.
+         *
+         * The proper format to map output power for an EV charging station is "<amount> kW",
+         * for example "22 kW" or "3.7 kW". Some fields in the wild are tagged with the unit "kVA"
+         * instead of "kW", those can be treated as equivalent.
+         *
+         * Sometimes people also mapped plain numbers (e.g. 7000, I assume that's 7 kW),
+         * ranges (5,5 - 11 kW, huh?) or even power (32 A), which is wrong. If we cannot parse,
```

Comment:
"`or even power (32 A)`" -> "`or even amperage (32 A)`"

Reply:
Ah, yeah, that's a mis-translation in my head 🙈